src/generate_linebreakdata.py

```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env python
+from __future__ import print_function
+import sys
+
+ranges = []
+last_first = None
+last_second = None
+last_value = None
+for line in sys.stdin:
+    first, second = line.rstrip().split('..')
+    second, value = second.split(';')
+    value = value.lstrip()
+    first = int(first, 16)
+    second = int(second, 16)
+    if last_value is None:
+        last_first = first
+        last_second = second
+        last_value = value
+    elif last_second + 1 != first or last_value != value:
+        ranges.append((last_first, last_second, last_value))
+        last_first = first
+        last_second = second
+        last_value = value
+    else:
+        last_second = second
+
+ranges.append((last_first, last_second, last_value))
+
+# print ranges for supplementary planes
+
+BMP = 65536
+
+sp = [r for r in ranges if r[0] >= BMP]
+
+print("""*/
+
+#include "linebreakdef.h"
+
+/** Line breaking properties for supplementary planes */
+const struct LineBreakProperties lb_prop_supplementary[{}] = {{""".format(len(sp) + 1))
+
+for first, second, value in sp:
+    if first >= BMP:
+        print("\t{{ 0x{:X}, 0x{:X}, LBP_{} }},".format(first, second, value))
+
+print("""\t{ 0xFFFFFFFF, 0xFFFFFFFF, LBP_Undefined }
+};
+""")
+
+# print BMP table
+
+print("""/** Line breaking properties for BMP */
+const char lb_prop_bmp[] = {""")
+
+bmp = ['XX'] * BMP
+for first, second, value in ranges:
+    if first >= BMP:
+        break
+    bmp[first:second + 1] = [value] * (second - first + 1)
+
+def chunks(arr, n):
+    for i in range(0, len(arr), n):
+        yield i, arr[i:i + n]
+
+for i, chunk in chunks(bmp, 16):
```

Comment:
`8` is better, as the code currently still can fit into the 80-column width.

BTW, there are a few PEP8 issues. But this is minor (fix them only if you want to).

Reply:
The removed Python script did not conform to PEP8, either. 🤪